src/index.ts

```diff
@@ -2,4 +2,11 @@ export { default as print } from "./print/print";
 export { default as getPrinters } from "./get-printers/get-printers";
 export { default as getDefaultPrinter } from "./get-default-printer/get-default-printer";
 export { PrintOptions } from "./print/print";
-export { Printer } from "./get-default-printer/get-default-printer";
+
+export type Printer = {
+  deviceId: string;
+  name: string;
+  paperSizes: string[];
+};
+
+export const properties: (keyof Printer)[] = ["deviceId", "name", "paperSizes"];
```

Comment:
that's the keys of Printer type, which is the keys of the final printerData object

Reply:
this is the last comment I have ;)

Reply:
> those properties will land in this command `Get-CimInstance Win32_Printer -Property ${props}`. So, the result will be `Get-CimInstance Win32_Printer -Property deviceId, name, paperSizes` and it is not going to work

oh you right, I missed that. How the tests suites were passing?

Reply:
before I had create this `properties` array and another object to map windows-keys to custom keys (PrinterPaperNames => paperSizes), now I changed `properties` to an object that map the keys. I think its better, because if we want to return any other keys, just add to this `properties` object and will return automatically.

```typescript
// map windows-printer key to final printerData key
export const properties: { [key: string]: keyof Printer } = {
  DeviceID: "deviceId",
  Name: "name",
  PrinterPaperNames: "paperSizes",
};
```

Reply:
no worries at all! i guess there are no tests for that ;)

please, keep the code simple, otherwise it takes me time to understand it xD. `const properties: string[] = ['DeviceID', 'Name', 'PrinterPaperNames']` will work fine. thanks!

Reply:
Ok, here are my povs:

Before I had created the following code:

```javascript
// map windows-printer key to custom key
const propertyMapping: { [key: string]: string } = {
  ["PrinterPaperNames".toLowerCase()]: "paperSizes",
};

// ...

const lowerLabel = label.toLowerCase();

const key =
  propertyMapping[lowerLabel] ??
  properties.find((prop) => prop.toLowerCase() === lowerLabel);
```

Ok, it was working, but if we had to add another property, we will need to add in `properties` array and here in `propertyMapping` (if the windows key was different of final printerData key, e.g `PrinterPaperNames` will be mapped to `paperSizes`).

So I changed to `properties` object, that map the Windows Keys to `printerData` Keys.

```javascript

// map windows-printer key to final printerData key
export const properties: { [key: string]: keyof Printer } = {
  DeviceID: "deviceId",
  Name: "name",
  PrinterPaperNames: "paperSizes",
};
```

```javascript
// send to Windows Get-CimInstance command

// before
const props = properties.join(",");

// after
const props = Object.keys(properties).join(",");
```

Once its typed with `{ [key: string]: keyof Printer }`, thats no way to had errors, because if we declare a new property in `Printer` type, that property need to be declared exact the same in `properties` object.

In my opinion, its more clean and readable, better to have an object `properties` rather than an array `properties` and an object `propertyMapping`. What do you think about that?

Reply:
I love keeping things simple :) In the current implementation, I need to go through various places to understand this line `Get-CimInstance Win32_Printer -Property ${props}`. Firstly, I need to find out what `props` are. It is created here `const props = Object.keys(properties).join(",")` and it is a list of something that is returned as a string. Then, I need to go to `index.ts` to understand what `properties` are, which is an object of keys that will be used in the string and values are some strings that probably used somewhere else too.
While it could just be one line of code `Get-CimInstance Win32_Printer -Property DeviceID,Name,PrinterPaperNames`. 
Additionally, I think exporting the `properties` object to users of pdf-to-printer may not be necessary.

Reply:
sorry for pushing back but I believe one line of code is better :)

Reply:
I understand your point, I just like to think far, thinking how to scalate this if we need adittional properties in the future, but thats ok.

I have replaced `const props = Object.keys(properties).join(",")` by directly passing `-Property DeviceID,Name,PrinterPaperNames`

I decided to keep `properties` private in `windows-printer-valid.ts` to set the values dynamically, instead of check every single key and set value, like was before:

```
// @ts-ignore
if (lowerLabel === "deviceid") printerData.deviceId = value;
```

I just want to you keep in mind, with these changes, if we had to add another property in the future, we have to declare in 

- `index.ts` in `Printer` type
- add the new property in `get-printers.ts` and `get-default-printer.ts`
- add the new property in `properties` in `windows-printer-valid.ts` 

while before, it was enough to add in `index.ts` file in `properties` object and `Printer` type.

just pushed my changes.